README.md

```diff
@@ -1 +1,119 @@
-# check-status 
+<!-- start title -->
+
+# gitgitgadget Action: check-status
+
+<!-- end title -->
+<!-- start description -->
+
+This is a GitHub action to perform various scheduled actions with projects using gitgitgadget to submit changes. It is used as a scheduled action running in a separate repo or the repo it is acting upon. The use case is for repos that will not allow complex GitHub actions in the codebase when the GitHub repo is a clone that is used to submit updates to a non-GitHub maintained repo.
+
+<!-- end description -->
+<!-- start contents -->
+<!-- end contents -->
+
+## Usage
+
+<!-- start usage -->
+
+```yaml
+- uses: gitgitgadget/check-status@v0.5.0
+  with:
+    # The action to be performed. It must be one of the following: "update-open-prs",
+    # "update-commit-mappings", "handle-open-prs", "handle-new-mails".
+    action: ""
+
+    # Repository owner.
+    repo-owner: ""
+
+    # Repository name.
+    repo-name: ""
+
+    # A repo scoped GitHub Personal Access Token.
+    token: ""
+
+    # The location of the repository.
+    repository-dir: ""
```

Comment:
Should we say that this is only needed if reusing an existing one on self-hosted runners?

Reply:
This is the equivalent of the `--git-work-dir` switch.  It is used to make workflows more generic.

Reply:
Right. What I was wondering is how we can make this as convenient to use as possible in the default case, i.e. as a GitHub Action that is run on build agents that have no persistent state.

I could imagine, for example, that we'll simply initialize a bare, empty repository in a hidden directory in the current directory and use _that_ if the user does not specify any `repository-dir`.

Reply:
> Right. What I was wondering is how we can make this as convenient to use as possible in the default case, i.e. as a GitHub Action that is run on build agents that have no persistent state.
> 
> I could imagine, for example, that we'll simply initialize a bare, empty repository in a hidden directory in the current directory and use _that_ if the user does not specify any `repository-dir`.

Don't some of the actions require the repo for analysis (not sure if everything is pulled)?  Some actions may only need the notes.  In the potential implementation with reusable workflows, the user is not specifying this value - the workflow is managing it.

Reply:
Right, let's not have an automagic fall-back.